ankaios_sdk/_components/workload.py

```diff
@@ -0,0 +1,738 @@
+        return self
+
+    def add_allow_rule(
+            self, operation: str, filter_masks: list[str]
+            ) -> "WorkloadBuilder":
+        """
+        Add an allow rule to the workload.
+
+        Args:
+            operation (str): The operation the rule allows.
+            filter_masks (list): The list of filter masks.
+
+        Returns:
+            WorkloadBuilder: The builder object.
+        """
+        self.allow_rules.append((operation, filter_masks))
+        return self
+
+    def add_deny_rule(
+            self, operation: str, filter_masks: list[str]
+            ) -> "WorkloadBuilder":
+        """
+        Add a deny rule to the workload.
+
+        Args:
+            operation (str): The operation the rule denies.
+            filter_masks (list): The list of filter masks.
+
+        Returns:
+            WorkloadBuilder: The builder object.
+        """
+        self.deny_rules.append((operation, filter_masks))
+        return self
+
+    def add_config(self, alias: str, name: str) -> None:
+        """
+        Link a configuration to the workload.
+
+        Args:
+            alias (str): The alias of the configuration.
+            name (str): The name of the configuration.
+        """
+        raise NotImplementedError("add_config is not implemented yet.")
+
+    def build(self) -> Workload:
+        """
+        Build the Workload object.
+        Required fields: workload name, agent name, runtime and
+        runtime configuration.
+
+        Returns:
+            Workload: The built Workload object.
+
+        Raises:
+            ValueError: If required fields are not set.
+        """
+        if self.wl_name is None:
+            raise ValueError("Workload can not be built without a name.")
+
+        workload = Workload(self.wl_name)
+
+        if self.wl_agent_name is None:
+            raise ValueError("Workload can not be built without an "
+                             + "agent name.")
+        if self.wl_runtime is None:
+            raise ValueError("Workload can not be built without a "
+                             + "runtime.")
+        if self.wl_runtime_config is None:
+            raise ValueError("Workload can not be built without a "
+                             + "runtime configuration.")
+
+        workload.update_agent_name(self.wl_agent_name)
+        workload.update_runtime(self.wl_runtime)
+        workload.update_runtime_config(self.wl_runtime_config)
+
+        if self.wl_restart_policy is not None:
+            workload.update_restart_policy(self.wl_restart_policy)
+        if len(self.dependencies) > 0:
+            workload.update_dependencies(self.dependencies)
+        if len(self.tags) > 0:
+            workload.update_tags(self.tags)
+        if len(self.allow_rules) > 0:
+            workload.update_allow_rules(self.allow_rules)
+        if len(self.deny_rules) > 0:
+            workload.update_deny_rules(self.deny_rules)
+
+        return workload
```

Comment:
There are 2 possible scenarios on how the user can get a Workload (at least this is what I implemented):
* Build one: use the Builder and create a workload from scratch. This will have by default only one mask, and that would be the `desiredState.workloads.<name>`.
* Get one from Ankaios: this will be generated using the `_from_proto` method and the masks will be empty. In case an update occurs on the workload, the correct mask will be added to the list.

Indeed, the builder uses the `update` methods of the Workload, but the mask is covered in the Workload implementation and it's calculated alright. The reason why the builder uses this method is that the value checks are written only once: in the workload class.